src/main/PlagueMain.java

```diff
@@ -519,35 +517,41 @@ public void init() {
             // @formatter:off
             if (
                 // plague team
-                t == Team.malis
-                // it isnt a vault
+                team == Team.malis
+                // it isn't a vault
                 || event.tile.block() != (isSerpulo ? Blocks.vault : Blocks.reinforcedVault)
                 // player is tapping other teams vault
-                || event.player.team() != t
+                || event.player.team() != team
             ) return;
             // (?)
-            float dist = new Vec2(t.core().tile.x, t.core().tile.y).dst(new Vec2(event.tile.x, event.tile.y)) * (float)15.0;
-            int cost = Mathf.clamp((int)snap(dist, (float)500f), 1000, 10000);
+            float dist = new Vec2(team.core().tile.x, team.core().tile.y).dst(new Vec2(event.tile.x, event.tile.y)) * (float)15.0;
+            // return items from vault to core
+            team.core().items.add(event.tile.build.items);
+            // Except for the thorium
+            team.core().items.remove(Items.thorium, event.tile.build.items.get(Items.thorium));
+            // how much out of 1000 aint been paid yet
+            int necessaryThorium = 1000 - event.tile.build.items.get(Items.thorium);
+
+            // how much extra expensive core is by distance
+            double costRatio = Mathf.clamp((int)snap(dist, (float)500f), 1000, 10000) / 1000.0;
+            // the payment yet due from core, gets less efficient to pay this way with distance
+            int cost = (int)(necessaryThorium * costRatio);
```

Comment:
this is just making it less efficient, using more network calls, and just being less efficient.